classes/core/PKPBladeViewServiceProvider.php

```diff
@@ -0,0 +1,267 @@
+use Illuminate\Support\Str;
+use Illuminate\Foundation\AliasLoader;
+use PKP\core\PKPString;
+
+class PKPBladeViewServiceProvider extends ViewServiceProvider
+{
+    public const VIEW_NAMESPACE_PATH = 'view\\components\\';
+
+    /**
+     * Boot the service provider
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        // Allow to render blade files as .blade e.g. with the .php extension
+        // but still allow to render views as .blade.php to accommodate default behavior.
+        View::addExtension('blade', 'blade');
+
+        AliasLoader::getInstance()->alias('Js', \Illuminate\Support\Js::class);
+
+        // This allows templates to be referenced explicitly, 
+        // e.g., @include('VIEW_NAMESPACE::some-template') or @include('VIEW_NAMESPACE::some-template'),
+        // or even allow to render view as view('VIEW_NAMESPACE::some-template', [....])
+        // which allow to render views from any namespace ambiguity and improving maintainability.
+        collect($this->app->get('config')->get('view.paths'))
+            ->each(fn ($path, $namespace) => view()->addNamespace($namespace, $path));
+        
+        // This allows to render components as <x-COMPONENT_NAMESPACE::some-component />
+        // which allow to render components from any namespace ambiguity and improving maintainability.
+        collect($this->app->get('config')->get('view.components.namespace'))
+            ->each(fn ($namespace, $prefix) => Blade::componentNamespace($namespace, $prefix));
+
+        // Use this macro to resolve the view path for a plugin class based component in the component class
+        // in the render method as `View:resolvePluginComponentViewPath($this, 'COMPONENTS_VIEW_PATH')`
+        View::macro(
+            'resolvePluginComponentViewPath',
+            function (\Illuminate\View\Component $component, string $viewPath): string {
+                $bladeCompiler = app()->get('blade.compiler'); /** @var \PKP\core\blade\BladeCompiler $bladeCompiler */
+                $classComponentNamespaces = $bladeCompiler->getClassComponentNamespaces();
+
+                $className = get_class($component);
+                $componentNamespace = substr($className, 0, strrpos($className, '\\'));
+                $pluginViewNamespace = array_search($componentNamespace, $classComponentNamespaces);
+
+                return "{$pluginViewNamespace}::{$viewPath}";
+            }
+        );
+        
+        // Register the @url directive compatible with Smarty url function,
+        // use as following in a blade view
+        // "@url([
+        //     'page' => 'citationstylelanguage',
+        //     'op' => 'get',
+        //     'path' => $citationStyle['id'],
+        //     'params' => $citationArgsJson
+        // ])"
+        Blade::directive('url', function ($expression) {
+            return "<?php
+                \$parameters = $expression ? (array) ($expression) : [];
```

Comment:
The use of `$expression` twice here will result in the array being constructed twice, which is probably not what we want. (Here and below.) Since we always expect an array to be passed in, let's just remove the test and cast.

Reply:
Right , I am only thinking about what has been passed not what will be compiled to before render .